controllers/middleware/metadb.go

```diff
@@ -0,0 +1,73 @@
+package middleware
+
+import (
+	"github.com/jinzhu/gorm"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"encoding/json"
+	"github.com/jaylevin/TMN-API/errs"
+)
+
+// Acts as a middle man to the http.HandlerFunc function
+type MetaDb struct {
+	Db       *gorm.DB
+	Recorder *httptest.ResponseRecorder
+}
+
+type Response struct {
+	Status int `json:"status"`
+	Response interface{} `json:"response"`
+}
+
+type MetaDbFunc func(recorder *httptest.ResponseRecorder, r *http.Request, db *gorm.DB) (error)
+
+func (this *MetaDb) Handler(f MetaDbFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, request *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+
+		var resp interface{} = nil
+
+		err := f(this.Recorder, request, this.Db)
+		if err != nil {
+			err := err.(*errs.Error)
+			this.Recorder.WriteHeader(err.Status())
+			resp = err.Root().Error()
+		} else if this.Recorder.Body.Bytes() != nil {
+			err := json.Unmarshal(this.Recorder.Body.Bytes(), &resp)
+			if err != nil {
+				log.Println("Error unmarshalling JSON: ", err)
+			}
+		}
+
+		// Write the response to buffer
+		w.Write(this.manufactureResponse(resp, request))
+
+		// Clear the buffer for next API request
+		this.Recorder = httptest.NewRecorder()
```

Comment:
There is most likely a smarter way of doing this. If someone wants to toy around with it- Basically, all the JSON responses (for any endpoint) are first written to a single pointer instance of a `httptest.ResponseRecorder`, which is stored in the `MetaDb.Recorder` struct. Then, the recorder's bytes are essentially just copied over to the http.ResponseWriter, which then constructs the http response.

This allows us to add middleware to all of our endpoints, such as the check for the `prettify` URL query in the `manufactureResponse` method.

A better approach would be to create a new response recorder instance every time an endpoint is reached, but the way that the Handler function is wrapped makes this difficult. I'm sure we'll think of something, but for now this works.